corehq/util/workbook_reading/adapters/csv.py

```diff
@@ -0,0 +1,47 @@
+from contextlib import contextmanager
+import csv
+import os
+
+from corehq.util.workbook_reading import (
+    SpreadsheetFileInvalidError,
+    SpreadsheetFileNotFound,
+    Workbook,
+)
+from .raw_data import make_worksheet
+from .utils import format_str_to_its_type
+
+
+@contextmanager
+def open_csv_workbook(filename):
+    try:
+        if os.stat(filename).st_size <= 1:
+            raise SpreadsheetFileInvalidError('File is empty')
+        with open(filename, "r") as csv_file:
+            yield _CSVWorkbookAdaptor(csv_file).to_workbook()
+    except (UnicodeDecodeError, csv.Error) as error:
+        # Rather than trying to determine why the file is not readable (invalid
+        # characters, file is encrypted, etc.), just raise an error that the file
+        # is not readable.
+        raise SpreadsheetFileInvalidError(str(error))
+    except IOError as error:
+        raise SpreadsheetFileNotFound(str(error))
+    except FileNotFoundError as error:
+        raise SpreadsheetFileNotFound(str(error))
+
+
+class _CSVWorkbookAdaptor(object):
+
+    def __init__(self, csv_file):
+        self._file = csv_file
+
+    def to_workbook(self):
+        rows = []
+
+        # Loop through the rows, and change any string to its appropriate type.
+        for row in csv.reader(self._file, delimiter=","):
+            formatted_row = []
+            for column in row:
+                formatted_row.append(format_str_to_its_type(column))
```

Comment:
I don't think it's a good idea to coerce types like this.  The xls and xlsx wrappers _do_ do some type coercion, but in those instances, the spreadsheet itself declares type information for each cell.  Here it's essentially forced to guess.  Additionally, the guessing happens independently for each cell in a column, so you could end up with non-homogeneous types in a column, which would be weird.

Case properties, custom user data, and custom location data are all user-defined fields, which we generally always handle as a string.  I believe they're all expected to be stored as strings too.  For numbers, it might not be an issue to coerce to a numeric type, then back to a string, though there could be subtle changes due to rounding.  This would also catch things like phone numbers, which may look like numbers, but shouldn't be treated as such.  Boolean types would be especially problematic.  If a user has a case property set to `"yes"`, then the app logic probably compares to the string `"yes"`.  This would coerce that to `True`, which would end up in the app as `"True"`.  Dates and datetimes are similarly an issue.  Users are often very particular about date formats, and this would normalize all of the to ISO 8601 standard.  Additionally, you can't know for sure whether a format is `M/D/YY` or `D/M/YY` without more information.

Other columns will correspond to CommCare-specific fields, which often _do_ have an expected type.  I'd expect it to be the responsibilty of the importer to do the coercion there, since the means of coercion will vary (eg should a date be stored as a date or a datetime?)